### ISSUE TITLE
Access to command line interface via TCP connection?

### ISSUE THREAD

**Issue by helander** [Yoshimi/yoshimi#205]
Is it possible to access the command line interface (or similar functionality) via a TCP connection?
I have made a simple but well working web interface for fluidsynth and want to create something similar for yoshimi, but currently I lack a method that allow me to connect to a relevant interface in yoshimi over a network.

**Comment by Ichthyostega** [comment]
Yoshimi has not built-in networking capabilities.
However, the CLI runs over STDIN / STOUT. So in the gist of the good old UNIX philosophy, you can use it as building block.

In fact I did so for another purpose: I have written a test driver and are gradually building up a test suite to ensure the sound is not altered inadvertently by ongoing development. In this setup, I forked a sub process with yoshimi connected via pipes. This way the test runner can control the application via CLI.

So basically what you describe _should be „feasible“_ ... the challenge might be rather that yoshimi is a quite complicated beast, and so is the CLI. However, possibly you just want to allow loading some presets from the standard distribution and let people hit some keys over the web...

**Comment by helander** [comment]
Hi Hermann and thank you very much for your feedback.

I am currently exploring various ways to ”connect” to yoshimi’s stdin/stdout.

Now I at least know that there is no builtin support for TCP connections, so thank you very much for that information.

I am looking at a rather simple web interface where you can control mostly volumes and intrument selections. For that I do not think that the command line interface is too complex :)

**Comment by abrolag** [comment]
On Thu, 11 Jul 2024 09:44:43 -0700
helander ***@***.***> wrote:

>Hi Hermann and thank you very much for your feedback.
>
>I am currently exploring various ways to ”connect” to yoshimi’s stdin/stdout.
>
>Now I at least know that there is no builtin support for TCP connections, so thank you very much for that information.
>
>I am looking at a rather simple web interface where you can control mostly volumes and intrument selections. For that I do not think that the command line interface is too complex :)
>

If it is only basic commands, one possible solution might be to use netjack (or
netjack2 if using jack2), then set Yoshimi to receive jack MIDI).

Details of Yoshimi's supported MIDI controls are in the User Guide embedded into
Yoshimi, "MIDI CCs and NRPNs" 

As well as that, you can MIDI learn any incoming CC or NRPN and build your own
set of controls.

-- 
Will J Godfrey {apparently now an 'elderly'}

**Comment by helander** [comment]
Thanks Will, but I assume that bank names and more importantly instrument names are not available via MIDI ?
It is a vital part for me to be able too look this information up from the running instance.

**Comment by abrolag** [comment]
On Thu, 11 Jul 2024 10:27:05 -0700
helander ***@***.***> wrote:

>Thanks Will, but I assume that bank names and more importantly instrument names are not available via MIDI ?
>It is a vital part for me to be able too look this information up from the running instance.

They not available to read via MIDI, but can be written. I didn't realise you
wanted to read them. For that you would indeed need to use the CLI.

-- 
Will J Godfrey {apparently now an 'elderly'}
